test/e2e/federated-service.go

```diff
@@ -340,11 +317,45 @@ func discoverService(f *framework.Framework, name string) {
 		if status.State.Terminated == nil {
 			return logerr(fmt.Errorf("container is not in terminated state"))
 		}
-		if status.State.Terminated.ExitCode == 0 {
+		if status.State.Terminated.ExitCode == code {
 			return nil
 		}
 
 		return logerr(fmt.Errorf("exited %d", status.State.Terminated.ExitCode))
-	}, time.Minute*2, time.Second*2).
-		Should(BeNil(), "%q should exit 0, but it never did", command)
+	}
+}
+
+func discoverService(f *framework.Framework, name string, exists bool) {
```

Comment:
I'd suggest adding a doc comment to explain what the parameters mean, especially exists.